extensions/_states/vault.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import absolute_import
+
+import logging
+
+import salt.utils
+
+log = logging.getLogger(__name__)
+
+try:
+    import hvac
+    DEPS_INSTALLED = True
+except ImportError:
+    log.debug('Unable to import the HVAC library.')
+    DEPS_INSTALLED = False
+
+__all__ = ['initialize']
+
+
+def __virtual__():
+    return DEPS_INSTALLED
+
+
+def initialized(name, secret_shares=5, secret_threshold=3, pgp_keys=None,
+               keybase_users=None, unseal=True):
+    ret = {'name': name,
+           'comment': '',
+           'result': '',
+           'changes': {}}
+    initialized = __salt__['vault.is_initialized']()
+
+    if initialized:
+        ret['result'] = True
+        ret['Comment'] = 'Vault is already initialized'
+    elif __opts__['test']:
+        ret['result'] = None
+        ret['comment'] = 'Vault will be initialized.'
```

Comment:
I recall seeing checks like this in salt (and it seems __opts__ just contains the [master/minion conf settings](https://docs.saltstack.com/en/latest/topics/development/dunder_dictionaries.html)?), but what's the reason for changing this behavior when `test` is set?